locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -836,6 +910,52 @@ protected JcloudsSshMachineLocation obtainOnce(ConfigBag setup) throws NoMachine
         }
     }
 
+    private void waitForWinRmAvailable(final NodeMetadata node, final LoginCredentials expectedCredentials, ConfigBag setup) {
+        // TODO: Reduce / remove duplication between this and waitForReachable
+        String waitForWinRmAvailable = setup.get(WAIT_FOR_WINRM_AVAILABLE);
+        checkArgument(!"false".equalsIgnoreCase(waitForWinRmAvailable), "waitForWinRmAvailable called despite waitForWinRmAvailable=%s", waitForWinRmAvailable);
+
+        long delayMs = -1;
+        try {
+            delayMs = Time.parseTimeString(""+waitForWinRmAvailable);
+        } catch (Exception e) {
+            // normal if 'true'; just fall back to default
+        }
+        if (delayMs<0)
+            delayMs = Time.parseTimeString(WAIT_FOR_WINRM_AVAILABLE.getDefaultValue());
+
+        // FIXME: remove this
+        LOG.info("Address: " + node.getPublicAddresses().iterator().next());
+        LOG.info("User: " + expectedCredentials.getUser());
+        LOG.info("Password: " + expectedCredentials.getOptionalPassword().get());
+        final Session session = WinRMFactory.INSTANCE.createSession(node.getPublicAddresses().iterator().next(),
+                expectedCredentials.getUser(), expectedCredentials.getOptionalPassword().get());
```

Comment:
Previously you have used `getOptionalPassword().orNull()` - can we assume it will exist here?